### PR TITLE
Simplified action handling

### PR DESCRIPTION
**Pull request by harshad1**
In this PR, I have reworked how text actions are handled.

- CommonTextActions has been removed
- Flow and logic has been simplified
- Lots of code has been moved from the individual file type actions to the base class

There should be no user-facing changes here, everything is a pure cleanup

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java [hunk 1]
```diff
@@ -238,25 +238,17 @@ public void simulateKeyPress(int keyEvent_KEYCODE_SOMETHING) {
...
-            setSelection(start + newCursorPos);
-        }
-    }
-
-    public int getShiftWidth(String text) {
-        if (text.contains("sw=2") || text.contains("shiftwidth=2")) {
-            return 2;
-        } else if (text.contains("sw=8") || text.contains("shiftwidth=8")) {
-            return 8;
-        } else {
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java:258]: getShiftWidth and any sw= mentions are just removed. right?

File: app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java [hunk 2]
```diff
@@ -605,31 +621,102 @@ protected boolean runCommonTextAction(final @StringRes int action) {
...
+            }
+            case R.string.tmaid_common_delete_lines: {
+                final int[] sel = StringUtils.getLineSelection(_hlEditor);
+                final Editable text = _hlEditor.getText();
+                final boolean lastLine = sel[1] == text.length();
+                final boolean firstLine = sel[0] == 0;
+                text.delete(sel[0] - (lastLine && !firstLine ? 1 : 0), sel[1] + (lastLine ? 0 : 1));
+                return true;
+            }
+            // case R.string.tmaid_markdown_end_line_with_two_spaces: {
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java:665]: big section of commented code

File: app/src/main/java/net/gsantner/opoc/util/StringUtils.java [hunk 3]
```diff
@@ -345,4 +347,35 @@ public static void selectLines(final EditText edit, final List<Integer> position
             edit.setSelection(selStart, selEnd);
         }
     }
+
+    // Search for matching pairs of backticks
+    // interpolate contents of backtick pair as SimpleDateFormat
+    public static String interpolateEscapedDateTime(final String snip) {
+        final StringBuilder interpolated = new StringBuilder();
+        final StringBuilder temp = new StringBuilder();
+        boolean isEscaped = false;
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/util/StringUtils.java:356]: that is part of the snippets PR, isn't it?

File: app/src/main/res/layout/quick_keyboard_button.xml [hunk 4]
```diff
@@ -12,16 +12,14 @@
 <ImageView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (gsantner) [inline app/src/main/res/layout/quick_keyboard_button.xml:12]: I don't know how you feel about it, but the TextAction-button worked really well & flawless since long time. I don't see the need to change something there. Do you?

**Comment by gsantner** [comment]
merge conflict, and some translations got in somehow wrongly. You can copy out the master values- folders and just override them on your branch.

File: app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java (hunk 1 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java:258]: Yeah. We don't use this anywhere. Shift is set per-file by long-pressing the indent button

File: app/src/main/java/net/gsantner/opoc/util/StringUtils.java (hunk 3 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/opoc/util/StringUtils.java:356]: I built this on top of the snippets PR because I knew there would be some conflicts otherwise. This PR is WIP partially because I am waiting to close the snippets work before cleaning up and working on this PR.

File: app/src/main/res/layout/quick_keyboard_button.xml (hunk 4 above)
Comment (harshad1) [reply app/src/main/res/layout/quick_keyboard_button.xml:12]: There is an issue with how variable padding interacts with the layout. This is an effort to correct it and make the button spacing change cleanly. But all of this is still experimental. I will look at it later, after snippets is done.

**Comment by gsantner** [comment]
So PR is in progress still right?

**Comment by harshad1** [comment]
Technically done. But I'd like to test for another day, unless you need to merge right away, for some reason...

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java [hunk 5]
```diff
@@ -77,21 +75,6 @@
 public class DocumentEditFragment extends GsFragmentBase implements TextFormat.TextFormatApplier {
     public static final String FRAGMENT_TAG = "DocumentEditFragment";
     public static final String SAVESTATE_DOCUMENT = "DOCUMENT";
-    private static final String SAVESTATE_CURSOR_POS = "CURSOR_POS";
-    private static final String SAVESTATE_PREVIEW_ON = "SAVESTATE_PREVIEW_ON";
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java:81]: We don't need to use the instance state, we are already tracking these with `_appSettings`. Removing redundant information

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java [hunk 6]
```diff
@@ -742,29 +736,14 @@ public void setUserVisibleHint(boolean isVisibleToUser) {
...
-                    _hlEditor.requestFocus();
-                }
-                _hlEditor.setSelection(lastPos);
-            } else if (_appSettings.isEditorStartOnBotttom()) {
-                if (!initPreview) {
-                    _hlEditor.requestFocus();
-                }
-                _hlEditor.setSelection(_hlEditor.length());
-            }
-        }
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java:761]: All of this logic has been simplified and moved directly into `onCreate`

File: app/src/main/java/net/gsantner/markor/activity/MainActivity.java [hunk 7]
```diff
@@ -407,7 +407,8 @@ public void onViewPagerPageSelected(int pos) {
                 @Override
                 public void onFsViewerConfig(FilesystemViewerData.Options dopt) {
                     dopt.descModtimeInsteadOfParent = true;
-                    dopt.rootFolder = getIntentDir(getIntent(), _appSettings.getFolderToLoadByMenuId(_appSettings.getAppStartupFolderMenuId()));
+                    dopt.rootFolder = _appSettings.getNotebookDirectory();
+                    dopt.startFolder = getIntentDir(getIntent(), _appSettings.getFolderToLoadByMenuId(_appSettings.getAppStartupFolderMenuId()));
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/MainActivity.java:411]: Unrelated fix - folder intent sometimes doesn't work. Still trying things

**Comment by gsantner** [comment]
I'm kinda sick and probably wont respond this week.

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java (hunk 5 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java:81]: What I would pursue in future is similar data class to dopt with props cursorPos, previewOn...etc. With read/write from json file. Instead of SharedPreferencesBackend i.e. HashMap backend with r/w from that json file.

But please even if you like it - don't make changes for that idea in this pull request, otherwise it's just growing :smile:

File: app/src/main/java/net/gsantner/markor/activity/MainActivity.java (hunk 7 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/activity/MainActivity.java:411]: #1626 #1636 #1635

File: app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java (hunk 2 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java:665]: can be removed ?

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java (hunk 6 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java:761]: hm, still everything works as expected when using some other applicatoins and switching back to markor? Especially on the main screen

**Comment by gsantner** [comment]
on the master there is a small change that once again fixes the issue with AutoFormatter too-early taking into affect, inserting wrongly stuff where it should not. this is about temporarly disabling auto format when loading file into editor.

**Comment by gsantner** [comment]
can i look into making some improvememts on this PR and maybe merge today?

or something bigger bad?

unrelated - what I noticed recently is, that filrmanager move/copy often crahes

**Comment by harshad1** [comment]
No. This PR is done (if we leave it open I will be tempted to keep tweaking something or the other) . Please make your changes and merge.

**Comment by harshad1** [comment]
> unrelated - what I noticed recently is, that filrmanager move/copy often crahes

Hmmm. Not good. I'll look into it
